lib/internal/url.js

```diff
@@ -1539,8 +1539,11 @@ function pathToFileURL(filepath, options = kEmptyObject) {
   const windows = options?.windows;
   if ((windows ?? isWindows) && StringPrototypeStartsWith(filepath, '\\\\')) {
     const outURL = new URL('file://');
+    // Handle extended UNC path and standard UNC path
     // UNC path format: \\server\share\resource
-    const hostnameEndIndex = StringPrototypeIndexOf(filepath, '\\', 2);
+    const isExtendedUNC = StringPrototypeStartsWith(filepath, '\\\\?\\UNC\\');
```

Comment:
IMO it'd be worth adding a link to Microsoft doc explaining why we're ignoring that part of the path.

Reply:
Thank you for your valuable opinion. I have included a comment regarding this matter.